pulp_rpm/app/serializers/prune.py

```diff
@@ -0,0 +1,81 @@
+from gettext import gettext as _
+
+from rest_framework import fields, serializers
+
+from pulp_rpm.app.models import RpmRepository
+
+from pulpcore.plugin.serializers import ValidateFieldsMixin
+from pulpcore.plugin.util import get_domain
+
+
+class PruneNEVRAsSerializer(serializers.Serializer, ValidateFieldsMixin):
+    """
+    Serializer for prune-old-NEVRAs operation.
+    """
+
+    repo_hrefs = fields.ListField(
+        required=True,
+        help_text=_(
+            "Will prune old RPMs from the specified list of repos. Use ['*'] to specify all repos."
+        ),
+        child=serializers.CharField(),
+    )
+
+    keep_days = serializers.IntegerField(
+        help_text=_(
+            "Prune NEVRAs introduced *prior-to* this many days ago. "
+            "Default is 14. A value of 0 implies 'keep latest NEVRA only.'"
+        ),
+        required=False,
+        min_value=0,
+        default=14,
+    )
+
+    repo_concurrency = serializers.IntegerField(
+        help_text=(
+            "Number of concurrent workers to use to do the pruning. "
+            "If not set then the default value will be used."
+        ),
+        allow_null=True,
+        required=False,
+        min_value=1,
+        default=10,
+    )
+
+    dry_run = serializers.BooleanField(
+        help_text=_(
+            "Determine what would-be-pruned and log the list of NEVRAs. "
+            "Intended as a debugging aid."
+        ),
+        default=False,
+        required=False,
+    )
+
+    def validate_repo_hrefs(self, value):
+        """
+        Check that repo_hrefs is not an empty list and contains either valid hrefs or "*".
+        Args:
+            value (list): The list supplied by the user
+        Returns:
+            The list of RpmRepositories after validation
+        Raises:
+            ValidationError: If the list is empty or contains invalid hrefs.
+        """
+        if len(value) == 0:
+            raise serializers.ValidationError("Must not be [].")
+
+        # prune-all-repos is "*" - find all RPM repos in this domain
+        if "*" in value:
+            if len(value) != 1:
+                raise serializers.ValidationError("Can't specify specific HREFs when using '*'")
+            return RpmRepository.objects.filter(pulp_domain=get_domain())
+
+        from pulpcore.plugin.viewsets import NamedModelViewSet
+
+        # We're pruning a specific list of RPM repositories.
+        # Validate that they are for RpmRepositories.
+        hrefs_to_return = []
+        for href in value:
+            hrefs_to_return.append(NamedModelViewSet.get_resource(href, RpmRepository))
```

Comment:
Um, hm. I have to refresh my memory on how get_resource() behaves if it can't find the resource. In any event - my take is to fail the call. But it would be "more polite" to check all the provided HREFs, deptermine which ones aren't RPM-repos, and fail with a message specifying *all* the incorrect repos, so the user can clean up their usage. WDYT?

Reply:
Current behavior is to fail with this error-msg:
```
(oci-env) (rpm_prune) ~/github/Pulp3/pulp-cli $ http POST :5001/pulp/api/v3/rpm/prune/ repo_hrefs:='["/pulp/api/v3/repositories/file/file/018fe883-4745-7531-9145-acd871cd1ec8/"]'
HTTP/1.1 400 Bad Request
{
    "repo_hrefs": [
        "URI /pulp/api/v3/repositories/file/file/018fe883-4745-7531-9145-acd871cd1ec8/ not found for rpmrepository."
    ]
}
```

I think it would be more polite to go through all the HREFs, catching the ones that fail. Then, if **any** fail, raise an error listing the failed ones, failing the call. Let me experiment some on that theme.